src/combinatorics/simple_graph/basic.lean

```diff
@@ -96,11 +96,15 @@ incident to `v`.
 lemma adj_iff_exists_edge {v w : V} :
   G.adj v w ↔ v ≠ w ∧ ∃ (e ∈ G.edge_set), v ∈ e ∧ w ∈ e :=
 begin
-  split, { intro, split, { exact G.ne_of_adj ‹_›, }, {use ⟦(v,w)⟧, simpa} },
+  split,
+  { intro,
+    split,
+    { exact G.ne_of_adj ‹_› },
+    { use ⟦(v,w)⟧, simpa } },
```

Comment:
```suggestion
  refine ⟨λ _, ⟨G.ne_of_adj ‹_›, ⟨⟦(v,w)⟧, _⟩⟩, _⟩,
  { simpa },
```

Reply:
If you want to golf further perhaps 
```
refine ⟨λ _, ⟨G.ne_of_adj ‹_›, ⟨⟦(v,w)⟧, by simpa⟩⟩, _⟩,
```
could work?

Reply:
This is just personal aesthetic preference, but `refine` is good for introducing and moving data around and setting subgoals. When you start nesting tactic calls in it gets messy. That said, I don't particularly care.

Reply:
Agreed, I'm happy either way.